### PR TITLE
Adding ideas from existing issues 

### LINKED ISSUE #16: 📝 Draft Project Idea: Impact Evaluation Indexer
**What is a Draft Project Idea?**

A Draft Project Idea is a concept for a project within the Optimism ecosystem that is generally considered valuable but has not yet been fleshed out into a clearly defined product. Draft Project Ideas may be converted into Ecosystem Project Ideas as they are explored further and more thoroughly specified.

---

## Summary
 
This is a subset of indexers in general. It'd be great to have a dedicated indexer that generates measurements of the impact of user behavior onchain. Think of this like a mix between an indexer and an algorithm that assigns value to certain user actions. Who are the most "impactful" users on chain?

## Possible Features

- Build on top of other indexers or build your own
- Create an algorithm that scores user impact onchain

### LINKED ISSUE #21: 🧪 Mad Science Project Idea: Onchain Movie
**What is an Mad Science Project Idea?**

A Mad Science Project Idea is an idea for a project within the Optimism ecosystem that may not have clear value but is generally cool as heck. Mad Science Project Ideas make great hackathon projects and tend to be a great way to explore Optimism, OP Mainnet, and the OP Stack.

---

## Summary

[The OP Stack](https://stack.optimism.io/) makes it possible to easily spin up new rollups. You could make an entire rollup that just exists to host a single movie. It's not clear why you would do this, but that's the point of Mad Science Project Ideas. Have fun!

## Possible Features

- Create an OP Stack chain that hosts a movie.
- Allow people to add comments or reviews about the movie.
- Create a custom video player that downloads the movie.

### LINKED ISSUE #25: 🧪 Mad Science Project Idea: Prediction Market Framework for Everything
**What is an Mad Science Project Idea?**

A Mad Science Project Idea is an idea for a project within the Optimism ecosystem that may not have clear value but is generally cool as heck. Mad Science Project Ideas make great hackathon projects and tend to be a great way to explore Optimism, OP Mainnet, and the OP Stack.

---

## Summary

[The OP Stack](https://stack.optimism.io/) makes it possible to easily spin up new rollups. You could create an L2 for prediction, voting, etc. that can tap into live elements everywhere like:
- Prediction markets on stats for other chains
- Predict who gets kicked off The Bachelor/Bachelorette
- Place predictions for fantasy sports outcomes

## Possible Features

- Create an OP Stack chain that acts like a prediction market for everything.

### LINKED ISSUE #26: 🧪 Mad Science Project Idea: Generalized "Airline Miles" Rollup
**What is an Mad Science Project Idea?**

A Mad Science Project Idea is an idea for a project within the Optimism ecosystem that may not have clear value but is generally cool as heck. Mad Science Project Ideas make great hackathon projects and tend to be a great way to explore Optimism, OP Mainnet, and the OP Stack.

---

## Summary

[The OP Stack](https://stack.optimism.io/) makes it possible to easily spin up new rollups. A common use-case pitched for crypto is bringing rewards points onchain. However, this tends to be too expensive on blockchains like Ethereum. To solve the cost problem, you could create a rollup entirely dedicated to rewards points. You could even create a framework for easily spinning up rewards programs hooked up to individual rollups.

## Possible Features

- Create an OP Stack chain that manages rewards points.
- Create a framework for launching OP Stack chains for rewards points.

### LINKED ISSUE #31: 🧪 Mad Science Project Idea: Bridge Lottery
**What is an Mad Science Project Idea?**

A Mad Science Project Idea is an idea for a project within the Optimism ecosystem that may not have clear value but is generally cool as heck. Mad Science Project Ideas make great hackathon projects and tend to be a great way to explore Optimism, OP Mainnet, and the OP Stack.

---

## Summary

Create a "slot machine" application for the bridge that randomly rewards depositors into the OP Mainnet Standard Bridge with an NFT or some other token. Chances of being rewarded could be entirely random or could be proportional to assets deposited.

## Possible Features

- Randomly reward users for depositing into the OP Mainnet Standard Bridge.
- Allow users to see their rewards.

### LINKED ISSUE #43: 📝 Draft Project Idea: Stop Snoozing!
**What is a Draft Project Idea?**

A Draft Project Idea is a concept for a project within the Optimism ecosystem that is generally considered valuable but has not yet been fleshed out into a clearly defined product. Draft Project Ideas may be converted into Ecosystem Project Ideas as they are explored further and more thoroughly specified.

---

## Summary

Have a hard time waking up every day? Need something to keep you accountable? Introducing: Stop Snoozing!

Stop Snoozing! is a concept for a simple smart contract application that incentivizes you to wake up every day before a specified time. You can use [bokkypoobah's Solidity DateTime library](https://github.com/bokkypoobah/BokkyPooBahsDateTimeLibrary) to make sure that a transaction is triggered on a set schedule within a certain time range. Users can put some ETH or ERC20 tokens up as stake that will be taken away if the transaction isn't sent on time. Easy!

## Possible Features

- Allow users to easily set up a wakeup schedule
- Allow users to have a different wakeup schedule on different days of the week
- Allow users to put up some stake in ETH or ERC20 tokens
- Require users to trigger a transaction every day according to the wakeup schedule
- Take away the stake if the user fails to trigger the transaction

## Impact

- Stop Snoozing! is a fun application that gets users familiar with making transactions every day
- Waking up is hard, but it's easier when you have a strong incentive to do so!

## Resources

- [Solidity DateTime library](https://github.com/bokkypoobah/BokkyPooBahsDateTimeLibrary)

### LINKED ISSUE #53: 🔴 Ecosystem Project Idea: Generated Contract Docs
**What is an Ecosystem Project Idea?**

An Ecosystem Project Idea is an idea for a project that someone could build within the Optimism ecosystem. As opposed to an Ecosystem Concept or an Ecosystem Idea, an Ecosystem Project is generally more fleshed out and corresponds to a relatively clear product to be built. 

---

## Summary

The [OP Stack](https://stack.optimism.io) contains various smart contracts that are currently held within the [contracts-bedrock](https://github.com/ethereum-optimism/optimism/tree/develop/packages/contracts-bedrock) package in the Optimism Monorepo. We'd like to make information about these smart contracts more accessible with an UI so that anyone looking to review the project can easily understand what's going on. This would be a static site that collects together automatically generated artifacts from the contracts-bedrock package and displays it in a digestible format.

In addition to the possible features noted below, feel free to mention any further artifacts that would improve the quality of the generated docs. These can be added to the monorepo.

## Possible Features

- List of contract names, addresses, and block explorer links to each contract
- Call graphs generated by [Slither](https://github.com/crytic/slither) or [Solidity Visual Auditor](https://consensys.net/diligence/blog/2019/10/solidity-visual-auditor-extension-for-vs-code/) to understand the relationships between contracts
- Dependency/inheritance graphs
- Slither detector output
- Code coverage detection
- Invariant docs
- Nicely rendered contract specs

## Impact

Better contract documentation makes contracts easier to review and helps auditors/whitehats more easily understand the system, which ultimately makes the OP Stack safer for its users.

### LINKED ISSUE #57: 🔴 Ecosystem Project Idea: subDAO Primitive
**What is an Ecosystem Project Idea?**

An Ecosystem Project Idea is an idea for a project that someone could build within the Optimism ecosystem. As opposed to an Ecosystem Concept or an Ecosystem Idea, an Ecosystem Project is generally more fleshed out and corresponds to a relatively clear product to be built. 

---

## Summary

[nintynick on Twitter](https://twitter.com/nintynick_/status/1659586048844877827), cofounder of [Hats Protocol](https://twitter.com/hatsprotocol), highlighted a missing primitive in the DAO space called the "subDAO". The subDAO structure was, in the tweet, defined as a structure for groups of 7-20 people with a level of baseline infrastructure that makes decision making easy without creating too much overhead.

## Potential Features

- Ideal for groups of 7-20 people
- Minimal voting weight infrastructure
- Ability to easily add/remove members
- All onchain
- NFT or attestation-based membership
- Ability to give some admin authorities to the DAO
- Voting-weight handled in a non-transferable way (no delegation)
- Ability to make voting weights immutable

## Impact

- Small groups of individuals need to be able to coordinate
- Current tools (multisigs) focus on very small groups with rigid requirements
- Other tools focus on larger groups
- Intermediate size is an underserved market

### LINKED ISSUE #65: 🔴 Ecosystem Project Idea: List of Crypto Legal Public Goods
**What is an Ecosystem Project Idea?**

An Ecosystem Project Idea is an idea for a project that someone could build within the Optimism ecosystem. As opposed to an Ecosystem Concept or an Ecosystem Idea, an Ecosystem Project is generally more fleshed out and corresponds to a relatively clear product to be built. 

---

## Summary

Various members of the Ethereum and broader crypto ecosystems have worked on public goods within the legal space over the years. A list of these resources would be highly valuable for anyone looking to better understand the current crypto legal landscape. These can range from opinions by legal experts to explicit legal frameworks and DAO tooling.

## Possible Features

- List out public goods in the crypto legal space
- Give a brief explainer of what the public good is
- Link to the work and any relevant authors

## Impact

- Crypto legal space is confusing and this may help people get a better understanding
- Help highlight projects for potential inclusion in future RPGF rounds

### LINKED ISSUE #66: 🔴 Ecosystem Project Idea: Audit Preparation Guide
**What is an Ecosystem Project Idea?**

An Ecosystem Project Idea is an idea for a project that someone could build within the Optimism ecosystem. As opposed to an Ecosystem Concept or an Ecosystem Idea, an Ecosystem Project is generally more fleshed out and corresponds to a relatively clear product to be built. 

---

## Summary

Audits are an important part of shipping smart contract software to production. Much of the value of an audit can be derived from the work that goes into preparing for the audit. A well-prepared audit can often lead to much better results than a hasty one.

A canonical audit preparation guide would be highly valuable to both Optimism and the broader Ethereum ecosystem. An ideal contributor for this task has significant experience with the auditing process and would be able to provide clear guidelines that anyone can follow to achieve the best audit results.

## Possible Features

- Explain when a project should be considering an audit
- Explain the type of testing that a project should have
- Explain the type of documentation that a project should have in general
- Explain the type of documentation that a project should be giving to an auditor
- Explain the audit process in general so that a developer can prepare for the nitty gritty details
- Describe situations that a developer should look out for during the audit process
- Explain failure modes in the audit process and how to avoid them
- Explain how to find the best auditor for a given audit
- Give an overview of the currently available auditing groups and the types of expertise they provide

## Impact

- Audits are expensive and a poorly prepared audit can be a huge waste of time and money
- Helps developers create safer and better-documented applications

### LINKED ISSUE #79: 🧪 Mad Science Project Idea: Simple Twitter with Attestations
**What is an Mad Science Project Idea?**

A Mad Science Project Idea is an idea for a project within the Optimism ecosystem that may not have clear value but is generally cool as heck. Mad Science Project Ideas make great hackathon projects and tend to be a great way to explore Optimism, OP Mainnet, and the OP Stack.

---

## Summary

This project involves creating a simple Twitter clone but with a twist - we'll be using attestations and the [AttestationStation](https://community.optimism.io/docs/identity/atst-v1/) smart contract. In this clone, usual Twitter interactions, such as liking, following, and tweeting, will be implemented as attestations. This project aims to demonstrate the functionality of attestations through a popular social media platform's model.

## Possible Features

- Implement the ability for users to 'like' tweets, represented as an attestation. This shows how a simple interaction on the platform can be modeled.
- Include the feature for users to retweet. This attestation captures the process of content sharing within the network.
- Implement 'following' other users as an attestation, this shows how relationships and interests can be represented.
- The ability to 'tweet' or post messages. Each tweet is an attestation, representing a content contribution by a user to the network.

## Impact

- This project is fun and educational. It provides a hands-on exploration of attestations.
- By implementing a Twitter-like platform using attestations, we demonstrate the power and versatility of attestations in a practical and relatable context.
- This project provides a solid foundation for further contributions. Other developers could build on it, either by adding more features or scaling it up.

### LINKED ISSUE #88: 🧪 Mad Science Project Idea: L2Geth Revamp
**What is an Mad Science Project Idea?**

A Mad Science Project Idea is an idea for a project within the Optimism ecosystem that may not have clear value but is generally cool as heck. Mad Science Project Ideas make great hackathon projects and tend to be a great way to explore Optimism, OP Mainnet, and the OP Stack.

---

## Summary

L2Geth was deprecated in favor of Bedrock last month. L2Geth, our legacy system, has two main parts: the Data Transport Layer (DTL) responsible for indexing data from Ethereum, and L2Geth itself, built on an older version of Geth. It would be interesting to have a new version of L2Geth that is based on the latest stable version of Geth. This might make syncing legacy OP Mainnet nodes faster and more reliable. You can continue to use the DTL or build your own consensus/execution client split—similar to how the OP Stack operates post-Bedrock.

## Possible Features

- Use of the latest stable release of Geth.
- Option to use the current DTL or develop a custom consensus client.
- Focus on L1 syncing, no need for Sequencer logic.
- No need for p2p logic.
- Rapid and efficient syncing of the legacy system, optimizing storage space.
- Should be able to correctly sync the legacy system, can compare to old legacy nodes.

## Impact

- Speeding up the sync process for anyone needing to access legacy data.
- Increasing the reliability of legacy nodes for those who keep them for archival purposes.

### PR DESCRIPTION
**Pull request by Nicca42**
# TL;DR

We are revamping the ideas list! This means we are moving all the project ideas from issues into markdown files. This will allow ups to have a gorgeous front end (built by @wit03 & co), making it much easier to use these ideas across hackathons, bounties and other developer-focused events. 

As there are over 100 ideas, they are being filtered before making it into the new site. There are three ways an idea gets filtered:
1. Added to the new site.
2. Closed because the idea was built. 
3. Closed because the idea does not fit in the new scope.

The new site will be going live in the coming weeks, so keep an eye on this repo! 

# Adding to new site:

The following issues are getting migrated to markdown files for better display and longevity. They have been broken down by category.

### Mad Science Project Idea:
* #96 
* #81 

### Ecosystem Project Ideas:
* #99 
* #98 
* #91 
* #83 
* #68 
* 

# Issues that were built 💖:

### The following ideas were built:

* #107 
    Closed by @xannyxs
* #12
    Closed by @shardulpilley, @JeanGuillemont
* #97 
    Closed by @LvisWang, @mattick27, @vaibhavchellani
* #94 
    Closed by @Maxservais, @lucas-op, @abhinav-woodstock 
* #90 
    Closed by @shuse2
* #85 
    Closed by RetroHunters
* #84 
    Closed by @Billy19191 & @wit03, @sergeybruska
* #67 
    Closed by @buildooor
* 

### The following ideas are being worked on/unclear if they have been completed:

If you have been tagged below, it is because you indicated you were working on the issue, **_but it is unclear if you finished it. If you have finished building it, please reply to this PR with the link to the finished build._** If you have not finished or do not reply, it will be assumed you did not finish, and the idea will be re-opened. 

* #100 
    @buildooor Please provide the link to the finished build here & in the issue.
* #95 
    @sambacha Please provide the link to the finished build here & in the issue.
* #92 
    @kevin-fruitful Please provide the link to the finished build here & in the issue.
* #86 
    @upnodedev Please provide the link to the finished build here & in the issue.
* #56 
    @Sh0g0-1758 Please provide the link to the finished build here & in the issue.
* #40 
    @brucexu-eth Please provide the link to the finished build here & in the issue.
* 

# Closed & out of scope: 

The following issues are no longer relevant/out of scope or otherwise do not contribute towards the vision for the Ideas List v2.

### Pre-filtered to close:

Closes #65 
Closes #57 
Closes #43 
Closes #16 
Closes #31 
Closes #26 
Closes #25 
Closes #21 

### Live filtered (to close): Mad Science Project Ideas:

Closes #88 
Closes #79 

### Live filtered (to close): Ecosystem Project Ideas:
Closes #66 
Closes #53

### CODE REVIEW

**Comment by Billy19191** [comment]
Hi, @Maxservais

We are working on this migration to the official which will get on Optimism.io domain. Markdown files will be stored on GitHub to make it as simple as possible to CRUD by foundation members. 

Thanks!
